packages/x-date-pickers/src/DateTimePicker/tests/DateTimePicker.test.tsx

```diff
@@ -3,18 +3,18 @@ import { expect } from 'chai';
 import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
 import { screen } from '@mui/internal-test-utils/createRenderer';
 import { createPickerRenderer, stubMatchMedia } from 'test/utils/pickers';
-import { pickersInputBaseClasses } from '@mui/x-date-pickers/PickersTextField';
 
 describe('<DateTimePicker />', () => {
   const { render } = createPickerRenderer();
 
-  it('should render in mobile mode when `useMediaQuery` returns `false`', () => {
+  it('should render in mobile mode when `useMediaQuery` returns `false`', async () => {
     const originalMatchMedia = window.matchMedia;
     window.matchMedia = stubMatchMedia(false);
 
-    render(<DateTimePicker />);
+    const { user } = render(<DateTimePicker />);
 
-    expect(screen.getByLabelText(/Choose date/)).to.have.class(pickersInputBaseClasses.input);
+    await user.click(screen.getByLabelText(/Choose date/));
+    expect(screen.getAllByRole('dialog')).to.have.length(1);
```

Comment:
Nitpick: You could use the following:
```suggestion
    expect(screen.getByRole('dialog')).toBeVisible();
```

Reply:
It "works", but not because `toBeVisible` does the right check.

If you don't open the picker, you want a clean error message, and that's not the case here :/
`getByRole` crashes when 0 element match:

```
TestingLibraryElementError: Unable to find an element with the role "dialog"
```

If you replace with `queryByRole`, then it's `toBeVisible` that crashes because it tries to access `style` of an un-existing element:

```
TypeError: Cannot read properties of null (reading 'style')
```

I'm not a fan of having tests fails because the tools used don't support a scenario.
Reading the test, you don't know if it was supposed to fail or not.

If you prefer, the following also works:

```
expect(screen.queryByRole('dialog')).to.not.equal(null);
```

Reply:
Fair points. 👍 
Your last suggestion sounds the most optimal and correct given the assertions we have access to. 💯